Sources/LeafKit/LeafToken.swift

```diff
@@ -23,19 +23,34 @@ public enum Operator: String, Equatable, CustomStringConvertible {
 }
 
 extension Operator {
-    var isBooleanOperator: Bool {
+    var isBoolean: Bool {
         switch self {
-        case .equals,
-             .notEquals,
-             .greaterThan,
-             .greaterThanOrEquals,
-             .lessThan,
-             .lessThanOrEquals,
-             .and,
-             .or:
-            return true
-        default:
-            return false
+            case .not,
+                 .equals,
+                 .notEquals,
+                 .greaterThan,
+                 .greaterThanOrEquals,
+                 .lessThan,
+                 .lessThanOrEquals,
+                 .and,
+                 .or:
+                return true
+            default:
+                return false
+        }
+    }
+    
+    var isBinary: Bool {
+        switch self {
+            case .not: return false
+            default: return true
+        }
+    }
+    
+    var isUnaryPrefix: Bool {
+        switch self {
+            case .not: return true
+            default: return false
         }
```

Comment:
Breaks observer properties into more granular considerations so building expressions during Parse can happen more easily - eg, `!one||!two` should correctly resolve into `(! one) || (! two)`